synapseclient/core/otel_config.py

```diff
@@ -0,0 +1,234 @@
+"""OpenTelemetry configuration for Synapse Python Client."""
+import os
+import sys
+from typing import Any, Dict, List, Optional
+
+from opentelemetry import metrics, trace
+from opentelemetry.exporter.otlp.proto.http.metric_exporter import OTLPMetricExporter
+from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter
+from opentelemetry.sdk.metrics import MeterProvider
+from opentelemetry.sdk.metrics.export import (
+    ConsoleMetricExporter,
+    PeriodicExportingMetricReader,
+)
+from opentelemetry.sdk.resources import (
+    SERVICE_INSTANCE_ID,
+    SERVICE_NAME,
+    SERVICE_VERSION,
+    Resource,
+)
+from opentelemetry.sdk.trace import SpanProcessor, TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor, ConsoleSpanExporter
+from opentelemetry.trace import Span, SpanContext, get_current_span
+
+# Default service name for the client
+DEFAULT_SERVICE_NAME = "synapseclient"
+CLIENT_VERSION = "unspecified"
+SYNAPSE_SERVICE_VERSION = f"synapse.{SERVICE_VERSION}"
```

Comment:
Perhaps not for this ticket, but the values here look similar also to the user-agent headers that we set so that information is available in the warehouse per processedaccessrecord.

Reply:
It is similar, and they serve a similar purpose. These resource level attributes let us group all requests from a particular user/service.